front/components/vaults/VaultManagedDatasourcesViewsModal.tsx

```diff
@@ -32,47 +32,58 @@ export default function VaultManagedDataSourcesViewsModal({
   systemVaultDataSourceViews,
   vault,
 }: VaultManagedDataSourcesViewsModalProps) {
+  const [systemDataSourceViews, vaultDataSourceViews] = useMemo(() => {
+    const [systemDataSourceViews, vaultDataSourceViews]: Record<
+      string,
+      DataSourceViewType
+    >[] = [{}, {}];
+    initialSelectedDataSources.forEach((dsv) => {
+      systemDataSourceViews[dsv.dataSource.sId] =
+        systemVaultDataSourceViews.find(
+          (sdsv) => sdsv.dataSource.sId === dsv.dataSource.sId
+        ) ?? dsv;
+      vaultDataSourceViews[dsv.dataSource.sId] = dsv;
+    });
+    return [systemDataSourceViews, vaultDataSourceViews];
```

Comment:
IMHO, a reduce would make the code easier to read.

Reply:
I did a reduce first but the code was longer and not very clearer